ecommerce_integrations/woocommerce/woocommerce_connection.py

```diff
@@ -0,0 +1,277 @@
+import base64
+import hashlib
+import hmac
+import json
+
+import frappe
+from frappe import _
+from frappe.utils import cstr
+
+from ecommerce_integrations.woocommerce.constants import MODULE_NAME, SETTINGS_DOCTYPE
+
+
+def verify_request():
+	woocommerce_settings = frappe.get_doc(SETTINGS_DOCTYPE)
+	sig = base64.b64encode(
+		hmac.new(
+			woocommerce_settings.secret.encode("utf8"), frappe.request.data, hashlib.sha256
+		).digest()
+	)
+	if (
+		frappe.request.data
+		and not sig == frappe.get_request_header("X-Wc-Webhook-Signature", "").encode()
+	):
+		frappe.throw(_("Unverified Webhook Data"))
+	frappe.set_user(woocommerce_settings.creation_user)
+
+
+@frappe.whitelist(allow_guest=True)
+def order(*args, **kwargs):
```

Comment:
If requests are `GET` then db.commit might be required? Can you confirm request type from WooCommerce? 

ERPNext integration had explicit commit, hence confirming if that wasn't required?

Reference: https://frappeframework.com/docs/user/en/api/database#database-transaction-model

Reply:
WooCommerce requests are POST type.